packit_service/worker/handlers/logdetective.py

```diff
@@ -0,0 +1,162 @@
+from packit_service.worker.mixin import ConfigFromEventMixin, PackitAPIWithDownstreamMixin
+from packit_service.worker.reporting import BaseCommitStatus
+from packit_service.worker.result import TaskResults
+
+logger = logging.getLogger(__name__)
+
+
+@reacts_to_as_fedora_ci(event=logdetective.Result)
+class DownstreamLogDetectiveResultsHandler(
+    FedoraCIJobHandler,
+    PackitAPIWithDownstreamMixin,
+    ConfigFromEventMixin,
+):
+    __test__ = False
+    task_name = TaskName.downstream_log_detective_results
+    check_name = "Log Detective Analysis"
+
+    def __init__(self, package_config: PackageConfig, job_config: JobConfig, event: dict):
+        super().__init__(package_config, job_config, event)
+
+        self.result = LogDetectiveResult.from_string(event.get("result", ""))
+        self.identifier = event.get("identifier", "")
+        self.log_detective_analysis_start = datetime.fromisoformat(
+            event.get("log_detective_analysis_start")
+        )
+        self.target_build = event.get("target_build")
+        self.build_system = event.get("build_system")
+        self._ci_helper: Optional[FedoraCIHelper] = None
+        self.branch_name = ""
+
+    @staticmethod
+    def get_checkers() -> tuple[type[Checker], ...]:
+        return (IsEventForJob,)
+
+    def run(self) -> TaskResults:
+        """Submit report about result of Log Detective analysis.
+
+        The status value is set to neutral when result matches `LogDetectiveResult.complete`,
+        because Log Detective is always analyzing a failed build, rather than checking
+        for a possible failure.
+        Information about Log Detective run, is retrieved and recorded state compared
+        to the change. If the state matches no further processing occurs.
+        Number of started runs is incremented and information about elapsed time recorded
+        by Pushgateway. New state of the run is then recorded.
+        """
+        logger.debug(f"Log Detective run {self.identifier} result: {self.result}")
+
+        if not self.project:
+            msg = f"No project set for Log Detective run: {self.identifier}"
+            logger.error(msg=msg)
+            return TaskResults(success=False, details={"msg": msg})
+
+        log_detective_run_model = LogDetectiveRunModel.get_by_identifier(identifier=self.identifier)
+        if not log_detective_run_model:
+            msg = f"Unknown identifier received from Log Detective: {self.identifier}"
+            logger.warning(msg=msg)
+            return TaskResults(success=False, details={"msg": msg})
+
+        if log_detective_run_model.status == self.result:
+            msg = f"Log Detective result for run {self.identifier} already processed"
+            logger.debug(msg)
+            return TaskResults(success=True, details={"msg": msg})
+
+        status = BaseCommitStatus.error
+        if self.result == LogDetectiveResult.complete:
+            status = BaseCommitStatus.neutral
```

Comment:
as I mentioned in the Slack discussion, I think we will need to conditionalise this as this currently maps to [error](https://github.com/packit/packit-service/blob/1fc2892134a2b6a39bb57effe9f4ae696b6eabee/packit_service/worker/reporting/enums.py#L39) for the non-Github forges (we use it mostly for things like "allowlisting failed" etc, so mapping to success wouldn't make sense)

Reply:
Done.